packages/core/lib/modules/event-manager.js

```diff
@@ -0,0 +1,57 @@
+import { EventEmitter, once } from 'events';
+
+
+export class EventManager {
+    eventEmitter: EventEmitter;
+    eventNames: string[];
+
+    constructor() {
+        this.eventEmitter = new EventEmitter();
+        this.eventNames = [];
+    }
+
+    registerEvent(eventName) {
+        if (this.getEventByName(eventName)) {
+            throw new Error('Event already exists');
+        }
+
+        this.eventNames.push(eventName);
+    }
+
+    registerEvents(eventsObject) {
+        Object.keys(eventsObject).forEach(key => this.registerEvent(eventsObject[key]));
+    }
+
+    getEventByName(eventName) {
+        return this.eventNames.find(e => e === eventName);
+    }
+
+    checkEvent(eventName) {
+        if (!this.getEventByName(eventName)) {
+            throw new Error(`Event with name "${eventName}" not found`);
+        }
+    }
+
+    emit(eventName, payload) {
+        this.checkEvent(eventName);
+        this.eventEmitter.emit(eventName, payload);
+        return this;
+    }
+
+    removeListener(eventName, callback) {
```

Comment:
how do I add a listener?

Reply:
need to use the .on()  or waitFor() methods

Reply:
ah, that wasn't obvious to me since neither of them take a callback directly